diff --git a/maestrowf/maestro.py b/maestrowf/maestro.py
index 40818dc..252b284 100644
--- a/maestrowf/maestro.py
+++ b/maestrowf/maestro.py
@@ -320,6 +320,16 @@ def run_study(args):
     # Copy the spec to the output directory
     shutil.copy(args.specification, study.output_path)
 
+    if args.entk or args.entk_store_workflow:
+        args.autono = True
+        from .backend_entk import Converter
+
+        my_path, my_dag = study.stage()
+        my_dag.set_adapter(batch)
+
+        conv = Converter(my_dag)
+        conv.process(args.entk_store_workflow)
+
     # Use the Conductor's classmethod to store the study.
     Conductor.store_study(study)
     Conductor.store_batch(study.output_path, batch)
@@ -422,6 +432,16 @@ def setup_argparser():
                      "studies (NOTE: breaks commands that use parameter labels"
                      " to search directories). [Default: %(default)s]")
 
+    # This is an arg for Maestro + Radical integration
+    run.add_argument("--entk", action="store_true", default=False,
+                     help="Will run Maestro far enough to get DAG "
+                     "then an external python script passes the data "
+                     "to Radical to launch the jobs")
+    run.add_argument("--entk_store_workflow", action="store_true", default=False,
+                     help="Does the same thing as the entk option and "
+                     "additionally creates a json file with the information "
+                     "that will be passed to Radical.")
+
     prompt_opts = run.add_mutually_exclusive_group()
     prompt_opts.add_argument(
         "-n", "--autono", action="store_true", default=False,
